ADAP-768: Update docs for materialized views/dynamic tables in v1.6 (#3935)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/postgres-configs.md

Reviewer: any idea why the diff is offset here? hard for me to compare the revision

Author: Not sure. I added the changelog block below the header, but it doesn't seem to be able to figure that out for some reason.

Author: I removed the changelog block and this is much clearer now. Since that's causing issues for git diff, let's add them in as a separate PR if we decide to do so.

<VersionBlock firstVersion="1.6">

> ## Materialized Views

Reviewer: i'm pretty sure the docs team's style guide only initialized the first word of a heading

```suggestion
## Materialized views
```

Author: Ah ok. I was trying to figure that out by looking at the headings and it seemed mixed.

<Changelog>

  - **v1.6.0:** Introduced support for `materialized_view`

</Changelog>

The Postgres adapter supports [materialized views](https://www.postgresql.org/docs/current/rules-materializedviews.html).
Indexes are the only configuration that is specific to `dbt-postgres`.

> The remaining configuration follows the general [materialized view](/docs/build/materializations#Materialized-View) configuration.

Author: Did I link to the `Materialized View` section of `materializations` correctly here?

Reviewer: i think heading anchors are lower case. the upper-case version didn't bring me to the heading we want
```suggestion
The remaining configuration follows the general [materialized view](/docs/build/materializations#materialized-view) configuration.
```

There are also some limitations that we hope to address in the next version.

### Monitored Configuration Changes

The settings below are monitored for changes applicable to `on_configuration_change`.

#### Indexes

Index changes (`CREATE`, `DROP`) can be applied without the need to rebuild the materialized view.
This differs from a table model, where the table needs to be dropped and re-created to update the indexes.
If the `indexes` portion of the `config` block is updated, the changes will be detected and applied
directly to the materialized view in place.

### Limitations

#### Changing materialization to and from "materialized_view"

Swapping an already materialized model to a materialized view, and vice versa, is not supported.
The workaround is to manually drop the existing materialization in the data warehouse prior to calling `dbt run`.

> Normally, re-running with the `--full-refresh` flag would resolve this, but not in this case.

Reviewer 2: ```suggestion
Running with `--full-refresh` flag will not work to drop the existing table or view and create the materialized view (and vice versa).
```

This would only need to be done once as the existing object would then be a materialized view.

> For example, assume a table, `my_model`, has already been materialized to the underlying data platform via `dbt run`.

Reviewer 2: ```suggestion
For example,`my_model`, has already been materialized as a table in the underlying data platform via `dbt run`.
```
